frontends/web/src/routes/exchange/exchange.tsx

```diff
@@ -202,39 +213,62 @@ export const Exchange = ({ code, accounts }: TProps) => {
-                      key={exchange.exchangeName}
-                      id={exchange.exchangeName}
-                      exchangeName={exchange.exchangeName}
-                      deals={exchange.deals}
-                      checked={selectedExchange === exchange.exchangeName}
-                      onChange={() => setSelectedExchange(exchange.exchangeName)}
-                      onClickInfoButton={setInfo}
-                    />
-                    ))}
-                  </div>
-
-                  {!noExchangeAvailable && <div className={style.buttonsContainer}>
-                    {supportedAccounts.length > 1 &&
-                    <Button
-                      className={style.buttonBack}
-                      secondary
-                      onClick={() => route('/buy/info')}>
-                      {t('button.back')}
-                    </Button>}
-                    <Button
-                      primary
-                      disabled={!selectedExchange}
-                      onClick={goToExchange} >
-                      {t('button.next')}
-                    </Button>
-                  </div>}
+                  {activeTab === 'buy' &&
+                    <>
+                      <div className={style.innerRadioButtonsContainer}>
+                        {noExchangeAvailable && (
+                          <p className={style.noExchangeText}>{t('buy.exchange.noExchanges')}</p>
+                        )}
+                        {!noExchangeAvailable && allExchangeDeals && allExchangeDeals.exchanges.map(exchange => exchange.supported && (
+                          <ExchangeSelectionRadio
+                            key={exchange.exchangeName}
+                            id={exchange.exchangeName}
+                            exchangeName={exchange.exchangeName}
+                            deals={exchange.deals}
+                            checked={selectedExchange === exchange.exchangeName}
+                            onChange={() => setSelectedExchange(exchange.exchangeName)}
+                            onClickInfoButton={setInfo}
+                          />
+                        ))}
+                      </div>
+                      {!noExchangeAvailable && <div className={style.buttonsContainer}>
+                        {supportedAccounts.length > 1 &&
+                            <Button
+                              className={style.buttonBack}
+                              secondary
+                              onClick={() => route('/buy/info')}>
+                              {t('button.back')}
+                            </Button>}
+                        <Button
+                          primary
+                          disabled={!selectedExchange}
+                          onClick={goToExchange} >
+                          {t('button.next')}
+                        </Button>
+                      </div>}
+                    </>
+                  }
```

Comment:
This living in `exchange.tsx` isn't ideal. IMO, we should also create a similar component like `sell.tsx` for this and call it `buy.tsx`. So here in `exchange.tsx` should have both `sell` and `buy` component separately and clearly defined.

We can then also elliminate the duplicated useEffects. I'll do all of the refactoring part once the BE is integrated.